### PR TITLE
fix(sdk): generated cli components has inconsistent styles after the style leak fix 

### LINKED ISSUE #49652: Generated CLI component has inconsistent styles after the style leak fix
The components generated from the Embedding CLI currently has incorrect/inconsistent styles, which is caused by the styling changes from the style leak fix. We should fix the generated components.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/49652

### Description

After we fixed the style leak issue in https://github.com/metabase/metabase/issues/44802, the generated CLI components looks really strange as we were relying heavily on the leaked styles (e.g. to apply the background color to the page's background). This PR restores the generated component's styles.

### How to verify

- Run the CLI locally. I usually use `docker rm -f metabase-enterprise-embedding && yarn add file:../metabase/resources/embedding-sdk && npx @metabase/embedding-sdk-react start` on an empty Vite project to run.
- The generated components should have a normal style again.

### Demo

Light Theme

CleanShot 2567-11-08 at 01 08 38@2x

Dark Theme

CleanShot 2567-11-08 at 01 08 46@2x

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts [hunk 1]
```diff
@@ -36,14 +44,15 @@ export const ANALYTICS_CSS_SNIPPET = `
...
+  text-decoration: none;
 }
 
 .dashboard-select {
   background: transparent;
   color: #509EE3;
   border: none;
   font-family: inherit;
-  font-size: 14px;
+  font-size: 16px;
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts:55]: This makes the `<select>` the same font size as other links.

CleanShot 2567-11-08 at 01 13 05@2x

File: enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts [hunk 2]
```diff
@@ -36,14 +44,15 @@ export const ANALYTICS_CSS_SNIPPET = `
 
 .analytics-header-right > a {
   color: #509EE3;
+  text-decoration: none;
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts:47]: We were relying on the leaked CSS reset.

File: enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts [hunk 3]
```diff
@@ -5,6 +5,14 @@ export const ANALYTICS_CSS_SNIPPET = `
   cursor: pointer;
 }
 
+.analytics-root.theme-light {
+  background: #F9FBFC;
+}
+
+.analytics-root.theme-dark {
+  background: #2D353A;
+}
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-css-snippet.ts:14]: Apply background color depending on the theme switcher's current theme.

File: enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-dashboard-snippet.ts [hunk 4]
```diff
@@ -23,53 +23,55 @@ import { AnalyticsContext } from "./analytics-provider"
...
   const [isCreateQuestion, toggleCreateQuestion] = useReducer((s) => !s, false)
 
   const isDashboard = !isCreateQuestion
 
   return (
-    <div className="analytics-container">
-      <div className="analytics-header">
-        <div>
-          ${userSwitcherEnabled ? "<UserSwitcher />" : ""}
+    <div className={\`analytics-root theme-\${themeKey}\`}>
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-dashboard-snippet.ts:34]: Use "hide whitespace changes" to view this diff. It basically just wraps the component with more CSS classes.

File: enterprise/frontend/src/embedding-sdk/cli/snippets/theme-switcher-snippet.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ const ICONS = {
     <svg viewBox="0 0 24 24">
       <path
         fill="none"
-        stroke="currentColor"
+        stroke="#2D353A"
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/cli/snippets/theme-switcher-snippet.ts:26]: restores the color of the theme switcher. it was showing as white, so it seems to disappear as it blends with the white background.

**Review by npretto — APPROVED** [review]

**Review by deniskaber — APPROVED** [review]

**Pull request merged**